### PR TITLE
[TFF-44] Notifications story

### PR DESCRIPTION
**Pull request by thofmann**
### Summary

Adds the automated email notification system.

### Checklist

- [x] Have you added unit and functional tests as appropriate?
- [x] Did you update/add documentation for new methods or changed functionality?
- [x] Have you merged the target branch down into this one?

### How to Review

1. Just review the code and tests. Manual testing is not particularly feasible for this story.

### Links

- [TFF-44](https://msoese.atlassian.net/browse/TFF-44)

### CODE REVIEW

File: src/lib/clock.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+//initialization code
+function getInitialTimeoutTime(){
+    let timestamp = moment.tz(Date.now(), 'America/Chicago');
+    let hour = parseInt(timestamp.format('H'));
+    // check for times past 18:00 (6:00pm)
+    if (hour >= 18) {
+        // increment to the next day
+        timestamp = timestamp.add(1, 'd');
+    }
+    timestamp.hour(18).minute(0).second(0);
```
Comment (mattmahn) [inline src/lib/clock.js]: If we had logs, the timestamp for sending emails would look _so_ good

File: src/lib/email.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    auth: {
+        user: process.env.GMAIL_USER,
+        pass: process.env.GMAIL_PASSWORD
+    }
+});
+
+export function sendMail(emails) {
+    let bcc = emails.join(',');
+    let subject = 'Tech Support Notification: Overdue Equipment';
+    let body = 'Hello,%0A%0AOur records indicate that you have overdue equipment that was checked out from Tech Support. Please visit us soon to return the equipment.%0A%0AThank you!';
```
Comment (mattmahn) [inline src/lib/email.js]: Nit: It would be nice to tell them which items are becoming overdue.

File: test/unit/lib/email.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+        let getStudentsWithOverdueItems = sinon.stub(StudentStore, "getStudentsWithOverdueItems");
+        getStudentsWithOverdueItems.returns([
+            {email: 'trevinhofmann@gmail.com'}
+        ]);
+        let spy = sinon.spy(email, 'sendMail');
+        return email.sendOverdueItemNotifications().then(() => {
+            assert.isTrue(spy.called, "Spy should be called");
+            StudentStore.getStudentsWithOverdueItems.restore();
+        });
+    }).timeout(15000);
```
Comment (mattmahn) [inline test/unit/lib/email.js]: That could also be done by not using a lambda, then using `this.timeout(15000)` in the callback function.

**Comment by thofmann** [comment]
@mattmahn 

>I don't see how TravisCI is not failing when there are no environment variables set.

They're set here: https://travis-ci.org/TheFourFifths/consus/settings

After a bit of research, it looks like that's more appropriate than in the `yml` file.

File: src/lib/email.js (hunk 2 above)
Comment (thofmann) [inline src/lib/email.js]: Agreed, except that this is one mass-email that will be send out, bcc'ing all students with overdue items. To send individual messages would be painful for the person doing it.

**Comment by mattmahn** [comment]
In that case, we need to have a grunt task that does not run these tests. Or have a config with the values inside...

**Comment by thofmann** [comment]
Why?

**Pull request merged**